plone/app/uuid/tests.py

```diff
@@ -59,6 +61,58 @@ def test_uuidToPhysicalPath(self):
         self.assertEqual('/'.join(d1.getPhysicalPath()),
                          uuidToPhysicalPath(uuid))
 
+    def test_speed(self):
+        # I updated some of the utility functions to be a bit faster.
+        # In this function you can check the speed.
+        from Acquisition import aq_base
+        from plone.uuid.interfaces import IUUID
+        from plone.app.uuid.utils import uuidToPhysicalPath
+        from plone.app.uuid.utils import uuidToURL
+        from plone.app.uuid.utils import uuidToObject
+        from plone.app.uuid.utils import uuidToCatalogBrain
+
+        portal = self.layer['portal']
+        setRoles(portal, TEST_USER_ID, ['Manager'])
+
+        start = time.time()
+        uuids = {}
+        total = 40
+        for i in range(total):
+            doc_id = portal.invokeFactory('Document', 'd{}'.format(i))
+            doc = portal[doc_id]
+            uuids[IUUID(doc)] = {
+                'path': '/'.join(doc.getPhysicalPath()),
+                'url': doc.absolute_url(),
+                'obj': aq_base(doc),
+            }
+        end = time.time()
+        print("Time taken to create {} items: {}".format(total, end - start))
+
+        self.assertEqual(len(uuids), total)
+        start = time.time()
+        for uuid, info in uuids.items():
+            self.assertEqual(info['path'], uuidToPhysicalPath(uuid))
+        end = time.time()
+        print("Time taken for uuidToPhysicalPath: {}".format(end - start))
```

Comment:
It is pretty unusual for the tests to print some output, see https://jenkins.plone.org/job/pull-request-6.0-3.8/756/consoleFull

Reply:
I have changed it to only print when an environment variable is set:

```
$ PLONE_APP_UUID_TEST_SPEED_TOTAL=100 bin/test -s plone.app.uuid -t test_speed
...
Creating 100 documents...
Time taken to create 100 items: 4.955342054367065
Time taken for uuidToPhysicalPath: 0.0012142658233642578
Time taken for uuidToURL: 0.05345273017883301
Time taken for uuidToObject: 0.02753305435180664
Time taken for uuidToCatalogBrain: 0.030633926391601562
```